Gems/RecastNavigation/Code/Source/Misc/RecastNavigationPhysXProviderCommon.h

```diff
@@ -41,6 +44,28 @@ namespace RecastNavigation
             const AZ::Aabb& worldVolume,
             bool debugDrawInputData);
 
+        //! Async variant of @CollectGeometryImpl. Tiles are returned via a callback @tileCallback.
+        //!   Calls on @tileCallback will come from a task graph (not a main thread).
+        //!   Is expected that the context for callback function will persist until all the tiles are processed.
+        //!
+        //! Note: the callback will be called at least once and one time for each tile geometry.
+        //! For example if two tiles are found the following callbacks will be made:
+        //!      tileCallback(tile1);
+        //!      tileCallback(tile2);
+        //!      tileCallback({}); //-- this indicates the end of the operation
+        //!
+        //! @param tileSize the result is packaged in tiles, which are squares covering the provided volume of @worldVolume
+        //! @param borderSize an additional extend in all direction around the tile volume, this additional geometry will allow Recast to connect tiles together
+        //! @param worldVolume worldVolume the overall volume to collect static PhysX geometry
+        //! @param debugDrawInputData debugDrawInputData if true, debug draw will show the geometry collected
+        //! @param tileCallback an empty tile indicates the end of the operation, otherwise a valid shared_ptr is returned with tile geometry
+        void CollectGeometryAsyncImpl(
+            float tileSize,
+            float borderSize,
+            const AZ::Aabb& worldVolume,
+            bool debugDrawInputData,
+            AZStd::function<void(AZStd::shared_ptr<TileGeometry>)> tileCallback);
```

Comment:
question: Is it possible to force this to run single threaded? Just for testing purposes if we need to debug something and check it's not to do with the async work? Maybe a CVAR? Or is there another way to do this, just curious as this could come in handy later potentially

Reply:
There is always blocking CollectGeometryImpl that runs on the main thread. Or setting the task thread count to 1 to simplify the testing.

Reply:
See `bg_navmesh_tileThreads`

Reply:
Just for clarity there are two APIs that do the same work: UpdateNavigationMeshBlockUntilCompleted and UpdateNavigationMeshAsync.

Reply:
Okay perfect thanks 👍